plugins/inputs/dpdk/dpdk.go

```diff
@@ -96,136 +82,236 @@ func (dpdk *dpdk) Init() error {
 		}
 
-		if commandWithoutParams := stripParams(commandWithParams); len(commandWithoutParams) >= maxCommandLength {
+		if commandWithoutParams := stripParams(cmd); len(commandWithoutParams) >= maxCommandLength {
 			return fmt.Errorf("%q command is too long. It shall be less than %v characters", commandWithoutParams, maxCommandLength)
 		}
 
-		if len(commandWithParams) >= maxCommandLengthWithParams {
-			return fmt.Errorf("command with parameters %q shall be less than %v characters", commandWithParams, maxCommandLengthWithParams)
+		if len(cmd) >= maxCommandLengthWithParams {
+			return fmt.Errorf("command with parameters %q shall be less than %v characters", cmd, maxCommandLengthWithParams)
 		}
 	}
 
 	return nil
 }
 
-// Gathers all unique commands and processes each command sequentially
-// Parallel processing could be achieved by running several instances of this plugin with different settings
-func (dpdk *dpdk) Gather(acc telegraf.Accumulator) error {
-	// This needs to be done during every `Gather(...)`, because DPDK can be restarted between consecutive
-	// `Gather(...)` cycles which can cause that it will be exposing different set of metrics.
-	commands := dpdk.gatherCommands(acc)
+// Establishes connections do DPDK telemetry sockets
+func (dpdk *dpdk) maintainConnections() error {
+	socketsToConnect, socketsToDisconnect := dpdk.identifySockets()
+
+	// Try to close connection with unused sockets
+	// And delete unused elements from the dpdk.connectors list with the safe approach
+	for _, socketToDisconnect := range socketsToDisconnect {
+		for i := 0; i < len(dpdk.connectors); i++ {
+			connector := dpdk.connectors[i]
+			if socketToDisconnect == connector.pathToSocket {
+				dpdk.Log.Debugf("Close unused connection: %s", socketToDisconnect)
+				if closeErr := connector.tryClose(); closeErr != nil {
+					dpdk.Log.Warnf("Failed to close unused connection - %v", closeErr)
+				}
+				dpdk.connectors = append(dpdk.connectors[:i], dpdk.connectors[i+1:]...)
+				i--
+				break
+			}
+		}
+	}
+
+	// Create connections
+	for _, socketToConnect := range socketsToConnect {
+		connector := newDpdkConnector(socketToConnect, dpdk.AccessTimeout)
+		connectionInitMessage, err := connector.connect()
+		if err != nil {
+			if dpdk.UnreachableSocketBehavior == unreachableSocketBehaviorError {
+				return fmt.Errorf("couldn't connect to socket %s: %w", socketToConnect, err)
+			}
+			dpdk.Log.Warnf("Couldn't connect to socket %s: %v", socketToConnect, err)
+			continue
+		}
+
+		dpdk.Log.Debugf("Successfully connected to the socket: %s. Version: %v running as process with PID %v with len %v",
+			socketToConnect, connectionInitMessage.Version, connectionInitMessage.Pid, connectionInitMessage.MaxOutputLen)
+		dpdk.connectors = append(dpdk.connectors, connector)
+	}
 
-	for _, command := range commands {
-		dpdk.processCommand(acc, command)
+	if len(dpdk.connectors) == 0 {
+		return fmt.Errorf("no active sockets connections present")
 	}
 
 	return nil
```

Comment:
How about also checking `dpdk.UnreachableSocketBehavior` here and log a warning. This way you can remove the `if` in `Start()` and other calls of `maintainConnections()`.

Reply:
Addressed in: https://github.com/influxdata/telegraf/pull/14308/commits/594fecb38d9f68e33ff03a50172ee132bc5811f6